namui/src/namui/hooks/ctx/render_ctx/public.rs

```diff
@@ -73,23 +63,78 @@ impl<'a> RenderCtx {
         RenderDone { rendering_tree }
     }
 
-    pub fn ghost_compose(&self, compose: impl FnOnce(&mut ComposeCtx)) -> RenderingTree {
-        self.disable_event_handling();
-        let rendering_tree = self.compose_inner(compose);
-        self.enable_event_handling();
+    /// Get RenderingTree but don't add it to the children.
+    pub fn ghost_compose(
+        &self,
+        compose: impl FnOnce(&mut ComposeCtx),
+        GhostComposeOption {
+            enable_event_handling,
+        }: GhostComposeOption,
+    ) -> RenderingTree {
+        let lazy: Arc<Mutex<Option<LazyRenderingTree>>> = Default::default();
+        {
+            let mut compose_ctx = ComposeCtx::new(
+                self.tree_ctx.clone(),
+                KeyVec::new_child(self.get_next_component_index()),
+                *self.matrix.lock().unwrap(),
+                self.renderer(),
+                lazy.clone(),
+                self.raw_event.clone(),
+                self.clippings.clone(),
+            );
+
+            let prev_enable_event = self.tree_ctx.enable_event_handling(enable_event_handling);
+
+            compose(&mut compose_ctx);
+
+            self.tree_ctx.enable_event_handling(prev_enable_event);
+        }
+        let rendering_tree = lazy.lock().unwrap().take().unwrap().into_rendering_tree();
         rendering_tree
     }
-    pub fn component(&self, component: impl Component) -> &Self {
-        self.add(
-            KeyVec::new_child(self.get_next_component_index()),
-            component,
-        );
-        self
+
+    /// Get RenderingTree but don't add it to the children.
+    pub fn ghost_component(
+        &self,
+        component: impl Component,
+        GhostComposeOption {
+            enable_event_handling,
+        }: GhostComposeOption,
+    ) -> RenderingTree {
+        let key = KeyVec::new_child(self.get_next_component_index());
+
+        let prev_enable_event = self.tree_ctx.enable_event_handling(enable_event_handling);
+
+        let rendering_tree = self.render_children(key, component);
+
+        self.tree_ctx.enable_event_handling(prev_enable_event);
+
+        rendering_tree
     }
+
     pub fn compose(&self, compose: impl FnOnce(&mut ComposeCtx)) -> &Self {
-        let rendering_tree = self.compose_inner(compose);
+        let rendering_tree = self.ghost_compose(
+            compose,
+            GhostComposeOption {
+                enable_event_handling: true,
+            },
+        );
         self.children.lock().unwrap().push(rendering_tree);
 
         self
     }
+    pub fn component(&self, component: impl Component) -> &Self {
+        let rendering_tree = self.ghost_component(
+            component,
+            GhostComposeOption {
+                enable_event_handling: true,
+            },
+        );
+        self.children.lock().unwrap().push(rendering_tree);
+        self
+    }
+}
+
+pub struct GhostComposeOption {
+    pub enable_event_handling: bool,
```

Comment:
What is this used for?
It's set up, but it doesn't seem to be used anywhere

Reply:
It used so clippy or cargo check doesn't catch that.

https://github.com/NamseEnt/namseent/pull/676/files#diff-72d3f427136795e439949f3f0a84ef2ea33d0b903b21388799c90d20ebd7f088R166

Reply:
Yes, the variable `enable_event_handling` is used.
But it is only for setting value. no one actually use this.


```rust
        let prev_enable_event = self.tree_ctx.enable_event_handling(enable_event_handling);

        let rendering_tree =
            self.renderer
                .render(key_vec, component, self.matrix, self.clippings.clone());

        self.tree_ctx.enable_event_handling(prev_enable_event);
```
Above is setting `enable_event_handling` to true, and render, then restore it.
But inside rendering, `enable_event_handling` is not used.

I guess the code that prevents event_handling seems to be missing according to the `enable_event_handling` value.
Can you check?

Reply:
You mean this? https://github.com/NamseEnt/namseent/pull/676/files#diff-4989163211ebf99f560b38bf8b1c0a34a2ca766e63aa3b8a297b01cd9e9667f2L33

Right

Reply:
Fixed